### ISSUE TITLE
v4 migration - dropdown's restructured, no .caret, no .dropdown

### ISSUE THREAD

**Issue by tshinnic** [twbs/bootstrap#17985]
Saw anomalous mentions of `.caret` and `.dropdown` in `bootstrap.css` and `_dropdown.scss`.  Realized they are leftover from the 3.x dropdown button usages.  The docs have changed, but migration doc needs to say much has changed for dropdown buttons.

Migration needs to mention dropdown's are constructed differently, `<div> <a>s` rather than `<ul><li><a>`.

Migration needs to mention that carets are now automatically appended to dropdown button texts, and so code does not need to include `<span class="caret">`.  

And of course the uses of `.caret` need to go away in scss/css, and maybe just some of the dropdown uses of `.dropdown` (there are other uses for 'navs'?).  Disturbing is seeing leftover bits like
```
        .btn > .caret,
        .dropup > .btn > .caret {
        .dropup .caret,
        .navbar-fixed-bottom .dropdown .caret {
        .btn .caret {
        .btn-lg .caret, .btn-group-lg > .btn .caret {
        .dropup .btn-lg .caret, .dropup .btn-group-lg > .btn .caret {
```
which makes one imagine bugs not yet discovered, esp. the third/fourth lines above.

**Comment by Startouf** [comment]
Also please add some references so that typing "caret" in the bootstrap can actually point us to information concerning the new implementations.

**Comment by mdo** [comment]
Fixed as part of #18768 I believe.
